r/tools/nixlibs.R

```diff
@@ -912,8 +912,6 @@ if (is_release) {
   VERSION <- VERSION[1, 1:3]
   arrow_repo <- paste0(getOption("arrow.repo", sprintf("https://apache.jfrog.io/artifactory/arrow/r/%s", VERSION)), "/libarrow/")
 } else {
-  # Don't override explictily set NOT_CRAN env var, as it is used in CI.
-  not_cran <- !env_is("NOT_CRAN", "false")
```

Comment:
This was why the macbuilder offline build was failing: because the submission was from the checkout and thus on a dev version, NOT_CRAN was being flipped, which meant that we weren't checking for online status.

Reply:
Thanks for cleaning that up

Reply:
```r
  # Don't override explictily set NOT_CRAN env var, as it is used in CI.
  not_cran <- !env_is("NOT_CRAN", "false")
``` 
Even if it adds a step (changing version) to testing dev checkouts on macbuilder I think it is still save (dev versions will never be on CRAN after all) and provides DX, especially for new contributors on not allow listed linux distros. But if you disagree then :shrug:

Reply:
I don't feel super strongly, but to argue for deleting it: it was also a poor developer experience trying to debug why the changes I was making weren't being picked up, why it appeared that env vars I hadn't set were somehow on. I'd also think that having more code paths that are different between a release and a development version is risky.

We do recommend that folks installing locally set `NOT_CRAN=true`, so it would be moot in that case. How about we leave it removed in this PR, and if it becomes a real issue with in the future, feel free to add something like this back then.